src/devices/machine/wd1000.h

```diff
@@ -107,11 +107,8 @@ class wd1000_device : public device_t
 
 	devcb_write_line m_intrq_cb, m_drq_cb;
 
-	struct
-	{
-		harddisk_image_device *drive;
-		uint16_t cylinder;
-	} m_drives[4];
+	harddisk_image_device *m_drives[4];
```

Comment:
Could you make this an `optional_device_array` while you're changing it anyway?  It makes it more declarative, and the C string hacking in device_start is ugly.

Reply:
Updated to used an `optional_device_array`, looks much cleaner, thanks.